cmd/opdoc/opdoc.go

```diff
@@ -400,29 +402,18 @@ func main() {
 	integerConstantsTableMarkdown(constants)
 	constants.Close()
 
-	txnfields := create("txn_fields.md")
-	fieldSpecsMarkdown(txnfields, logic.TxnFieldNames, logic.TxnFieldSpecByName)
-	txnfields.Close()
-
-	globalfields := create("global_fields.md")
-	fieldSpecsMarkdown(globalfields, logic.GlobalFieldNames, logic.GlobalFieldSpecByName)
-	globalfields.Close()
-
-	assetholding := create("asset_holding_fields.md")
-	fieldSpecsMarkdown(assetholding, logic.AssetHoldingFieldNames, logic.AssetHoldingFieldSpecByName)
-	assetholding.Close()
-
-	assetparams := create("asset_params_fields.md")
-	fieldSpecsMarkdown(assetparams, logic.AssetParamsFieldNames, logic.AssetParamsFieldSpecByName)
-	assetparams.Close()
-
-	appparams := create("app_params_fields.md")
-	fieldSpecsMarkdown(appparams, logic.AppParamsFieldNames, logic.AppParamsFieldSpecByName)
-	appparams.Close()
-
-	acctparams, _ := os.Create("acct_params_fields.md")
-	fieldSpecsMarkdown(acctparams, logic.AcctParamsFieldNames, logic.AcctParamsFieldSpecByName)
-	acctparams.Close()
+	written := make(map[string]bool)
+	opSpecs := logic.OpcodesByVersion(logic.LogicVersion)
+	for _, spec := range opSpecs {
+		for _, imm := range spec.Details.Immediates {
+			if imm.Group != nil && !written[imm.Group.Name] {
+				out := create(imm.Group.Name + "_fields.md")
+				fieldSpecsMarkdown(out, imm.Group.Names, imm.Group.Specs)
+				out.Close()
+				written[imm.Group.Name] = true
+			}
+		}
+	}
 
 	langspecjs := create("langspec.json")
```

Comment:
@jannotti I'm early in the PR review, so there's a number of things I don't yet follow.  Though I discovered a difference that feels ambiguous to me.  Posting before I lose the note.  What do you think?

`langspec.json` generation varies for `ecdsa_verify`.  Here's output showing the delta generated by invoking `cmd/opdoc` on each branch:

```
~/dev/go-algorand @ca7504fb ?28 ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── 11:50:45 AM
❯ cat langspec.json.master | python -m json.tool | grep -A 5 "ecdsa_verify" | grep Cost
            "Cost": 1700,

~/dev/go-algorand @ca7504fb ?28 ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────── 11:50:54 AM
❯ cat langspec.json.field-costs | python -m json.tool | grep -A 5 "ecdsa_verify" | grep Cost
            "Cost": 0,
```

I don't know enough yet to know if it's _expected_ to go from non-zero to zero cost.

Reply:
It's "intentional" in the sense that Cost can't be described succinctly anymore, but my intent was to remove it, not have it show up as 0.  I'm told we don't use the Cost field in our SDKs, and we've checked with various partners and though don't either.

Reply:
I'm going to add langspec.json and teal.tmLanguage.json to git, so that changes like this are more obvious.